worker/queueservice.go

```diff
@@ -85,10 +87,11 @@ func (q *queueService) Start() <-chan *taskClaim {
 
 	go func() {
 		for !q.halt.Get() {
-			q.mu.RLock()
-			capacity := q.capacity
-			q.mu.RUnlock()
-			tasks := q.retrieveTasksFromQueue(capacity)
+			claimSize, finished := q.claimSize()
+			if finished {
+				break
```

Comment:
Shouldn't it just skip claiming? I might be wrong here... ie. Sleep a second and then `continue`.

Ideally, we probably should be doing Cond.Wait or something, but let's not complicate with that yet...